ducktape/command_line/main.py

```diff
@@ -189,7 +194,26 @@ def main():
         session_logger.warning("specified number of deflake runs specified to be less than 1, running without deflake.")
     deflake_num = max(1, deflake_num)
     runner = TestRunner(cluster, session_context, session_logger, tests, deflake_num)
-    test_results = runner.run_all_tests()
+
+    # Detect hung tests
+    test_results: Optional[TestResults] = None
+    try:
+        test_results = runner.run_all_tests()
+    except TimeoutError as e:
+        print(f"Timeout: assuming hung test.\n{e}")
```

Comment:
is this a safe assumption? I think its generally probably going to be the case, but shouldn't this be handled on the test_run client level to check if a test itself is hung?

Reply:
Based on my search of the code at the time, I think so. Should be clearer when moved to the runner, though, thanks.